src/components/pagination-nav/pagination-nav.vue

```diff
@@ -0,0 +1,169 @@
+<template>
+  <nav
+    role="navigation"
+    :aria-label="$t('pagination_navigation')"
+  >
+    <ul class="pagination-nav__list body">
+      <li class="pagination-nav__item">
+        <app-link
+          v-if="props.currentPage > 1"
+          :to="toLink(previousPage)"
+          :aria-label="$t('previous_page')"
+        >
+          &lt; {{ $t('previous') }}
+        </app-link>
+      </li>
+      <li
+        v-for="page in paginationRange"
+        :key="page"
+        class="pagination-nav__item"
+        :class="{
+          'pagination-nav__item--active': page === props.currentPage,
+        }"
+      >
+        <app-link
+          v-if="page === props.currentPage"
+          :to="toLink(page)"
+          :aria-label="$t('current_page', { page })"
+          aria-current="true"
+        >
+          {{ page }}
+        </app-link>
+        <app-link
+          v-else-if="page !== DOTS"
+          :to="toLink(page)"
+          :aria-label="$t('go_to_page', { page })"
+        >
+          {{ page }}
+        </app-link>
+        <span
+          v-else
+          class="pagination-nav__item"
+          aria-hidden="true"
+        >
+          {{ DOTS }}
+        </span>
+      </li>
+      <li class="pagination-nav__item">
+        <app-link
+          v-if="props.currentPage < totalPages"
+          :to="toLink(nextPage)"
+          :aria-label="$t('next_page')"
+        >
+          {{ $t('next') }} >
+        </app-link>
+      </li>
+    </ul>
+  </nav>
+</template>
+
+<script setup>
+  const DOTS = '...'
+
+  const props = defineProps({
+    totalItems: {
+      type: Number,
+      required: true,
+    },
+    currentPage: {
+      type: Number,
+      required: true,
+    },
+    perPage: {
+      type: Number,
+      required: true,
+    },
+  })
+
+  const totalPages = computed(() => Math.ceil(props.totalItems / props.perPage))
+  const previousPage = computed(() => props.currentPage - 1)
+  const nextPage = computed(() => props.currentPage + 1)
+
+  const paginationRange = computed(() => {
+    const siblingCount = 1
+    const firstPage = 1
+    const lastPage = totalPages.value
+    // Pages count is determined as siblingCount + firstPage + lastPage + currentPage + 2*DOTS
+    const totalPageNumbers = 2 * siblingCount + 5
```

Comment:
I don't understand this comment, does it relate to the `totalPageNumbers` definition?

Reply:
The comment is there to explain where the magic 5 comes from, I hope is more clear now.